Shared/Realm.Shared/RealmCollectionBase.cs

```diff
@@ -176,6 +176,44 @@ private void OnChange(IRealmCollection<T> sender, ChangeSet change, Exception er
             }
             else if (change != null)
             {
+                if (change.Moves.Length > 0 && 
```

Comment:
I'm not a big fan of assertions - they are useless for our users and we have unit tests to assert correct behavior.